src/test/java/io/pravega/connectors/flink/table/catalog/pravega/formats/pravega/registry/PravegaRegistryFormatFactoryTest.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.pravega.connectors.flink.table.catalog.pravega.formats.pravega.registry;
+
+import org.apache.flink.api.common.serialization.DeserializationSchema;
+import org.apache.flink.api.common.serialization.SerializationSchema;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.catalog.CatalogTableImpl;
+import org.apache.flink.table.catalog.ObjectIdentifier;
+import org.apache.flink.table.connector.sink.DynamicTableSink;
+import org.apache.flink.table.connector.source.DynamicTableSource;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.flink.table.factories.TestDynamicTableFactory;
+import org.apache.flink.table.runtime.connector.source.ScanRuntimeProviderContext;
+import org.apache.flink.table.runtime.typeutils.InternalTypeInfo;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.util.TestLogger;
+import org.junit.Test;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+/** Tests for the {@link PravegaRegistryFormatFactory}. */
+public class PravegaRegistryFormatFactoryTest extends TestLogger {
+
+    private static final TableSchema SCHEMA =
+            TableSchema.builder()
+                    .field("a", DataTypes.STRING())
+                    .field("b", DataTypes.INT())
+                    .field("c", DataTypes.BOOLEAN())
+                    .build();
+
+    private static final RowType ROW_TYPE = (RowType) SCHEMA.toRowDataType().getLogicalType();
+
+    private static final String SCOPE = "test-scope";
+    private static final String STREAM = "test-stream";
+    private static final URI SCHEMAREGISTRY_URI = URI.create("http://localhost:10092");
+
+    @Test
+    public void testSeDeSchema() {
+        final PravegaRegistryRowDataDeserializationSchema expectedDeser =
+                new PravegaRegistryRowDataDeserializationSchema(ROW_TYPE, InternalTypeInfo.of(ROW_TYPE),
+                        SCOPE, STREAM, SCHEMAREGISTRY_URI);
+
+        final Map<String, String> options = getAllOptions();
+
+        final DynamicTableSource actualSource = createTableSource(options);
+        assert actualSource instanceof TestDynamicTableFactory.DynamicTableSourceMock;
```

Comment:
Try to use
```
assertThat(xxx, instanceOf(yClass.class));
```
ref: https://stackoverflow.com/questions/12404650/assert-an-object-is-a-specific-type